src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1003,15 +1058,12 @@ private void onDecodePublish(
                 final MessageConsumer newTarget = router.supplyReceiver(newInitialId);
 
                 final int topicKey = topicKey(topicName);
-                MqttServer.MqttPublishStream publishStream = publishers.computeIfAbsent(topicKey, s ->
-                {
-                    final MqttPublishStream newPublishStream = new MqttPublishStream(newTarget,
-                                                                     newRouteId, newInitialId, newReplyId, 0, topicName);
-                    newPublishStream.doApplicationBegin(traceId, authorization, affinity);
-                    return newPublishStream;
-                });
-                publishStream.doApplicationData(traceId, authorization, payload, dataEx);
-                correlations.put(newReplyId, publishStream);    // TODO: do we need to clean up correlations onAbort()?
+                MqttServerStream stream = streams.computeIfAbsent(topicKey, s ->
+                                                        new MqttServerStream(newTarget, newRouteId, newInitialId, newReplyId,
+                                                            0, topicName, MqttRole.SENDER));
```

Comment:
Please move `newInitialId`, `newTarget` and `newReplyId` derivations into `MqttServerStream` constructor - only requires `newRouteId` to derive those.